### PR TITLE
ci(actions): add backmerge action

### PR DESCRIPTION
**Pull request by dnlup**
## 🚨 Proposed changes

> Please review the [guidelines for contributing](../../CONTRIBUTING.md) to this repository.

[[Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.]]

## ⚙️ Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)
-   [ ] Refactor

In this MR we add an action to perform the back merge into the `next` branch since the `semantic-release` plugin is not working.

### CODE REVIEW

File: .github/workflows/backmerge.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Backmerge main -> next
+
+on:
+    push:
+        branches:
+            - 'main'
```
Comment (dnlup) [inline .github/workflows/backmerge.yml:6]: Maybe we can use the [release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) event?

Comment (simonecorsi) [reply .github/workflows/backmerge.yml:6]: LGTM! We can avoid useless run!

**Comment by antoniomuso** [comment]
Have we decided to use a separated action to do back merge?

**Comment by simonecorsi** [comment]
> Have we decided to use a separated action to do back merge?

What I don't like about the current implementation is that we're tied to an external plugin and perfectly fine release jobs fails in the aftermath just because of it.
 
We can't update `semantic-release` until the owner updates it to ESM.

I think we can split these two job and trigger the backmerge on `release` as @dnlup suggested, we would have the same result but they would be "atomic", not in the strict sense, but in the sense that we could run and retry both separately.

File: .github/workflows/backmerge.yml (hunk 1 above)
Comment (dnlup) [reply .github/workflows/backmerge.yml:6]: I was thinking, we have to be able to separate `releases` from `prereleases`, it should be handled by the `release` and `pre-release` event but I am not sure 🤔

Comment (dnlup) [reply .github/workflows/backmerge.yml:6]: I think we need to add a check on the type `released` in the job

**Review by dnlup — COMMENTED** [review]
I think we also need to remove the `semantic-release` plugin.

File: .github/workflows/backmerge.yml (hunk 1 above)
Comment (simonecorsi) [reply .github/workflows/backmerge.yml]: I think the `released` action type won't get `pre-release` they also have a different [tag](https://github.com/immobiliare/fastify-metrics/releases/tag/v2.0.0-next.3) looking at documentation seems like to me that if you want both regardless of type you should use `published`, at least that's my understanding

File: .github/workflows/backmerge.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+              with:
+                  persist-credentials: false
+
+            - name: Merge main -> next
+              uses: devmasx/merge-branch@v1.4.0
+              with:
+                  type: now
+                  from_branch: main
+                  target_branch: next
+                  github_token: ${{ secrets.GITHUB_TOKEN  }}
```
Comment (simonecorsi) [inline .github/workflows/backmerge.yml:22]: `GH_NODE_TOKEN`

**Review by simonecorsi — CHANGES_REQUESTED** [review]

File: .github/workflows/backmerge.yml (hunk 2 above)
Comment (simonecorsi) [reply .github/workflows/backmerge.yml:22]: No, it's not needed on every job, just jobs that have to perform write operations

**Review by simonecorsi — APPROVED** [review]
LGTM 🚀

**Pull request merged**